File: mcp-server/internal/domains/mcpservers/resources.go
package mcpservers

import (
	"github.com/modelcontextprotocol/go-sdk/mcp"

	"github.com/stigmer/stigmer/mcp-server/internal/domains"
)

// Template returns the MCP resource template for MCP servers.
func Template() *mcp.ResourceTemplate {
	return &mcp.ResourceTemplate{
		URITemplate: "stigmer://mcp-servers/{org}/{slug}",
		Name:        "stigmer_mcp_server",
		Title:       "Stigmer MCP Server",
		Description: "Full definition of a Stigmer MCP server, identified by organization and slug.",
		MIMEType:    "application/json",
	}
}

// ResourceHandler returns a handler that reads an MCP server resource by
// parsing the org and slug from the request URI.
func ResourceHandler(serverAddress string) mcp.ResourceHandler {
	return domains.NewResourceHandler(Fetch, serverAddress, "mcpservers")
}

File: mcp-server-ts/src/domains/mcpservers/resources.ts
// MCP server resource template (stigmer://mcp-servers/{org}/{slug}).
// Go parity: mcp-server/internal/domains/mcpservers/resources.go.

import type { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";

import type { BackendTarget } from "../client";
import { registerResource } from "../resourcehandler";
import { fetchMcpServer } from "./fetch";

/** Register the MCP server resource template; returns the registered resource names. */
export function registerMcpServerResources(server: McpServer, target: BackendTarget): string[] {
  registerResource(server, target, {
    name: "stigmer_mcp_server",
    title: "Stigmer MCP Server",
    description: "Full definition of a Stigmer MCP server, identified by organization and slug.",
    template: "stigmer://mcp-servers/{org}/{slug}",
    fetch: fetchMcpServer,
  });
  return ["stigmer_mcp_server"];
}
